blocks/api/paste.js

```diff
@@ -76,7 +78,9 @@ export function normaliseToBlockLevelNodes( nodes ) {
 }
 
 export default function( nodes ) {
-	return normaliseToBlockLevelNodes( nodes ).map( ( node ) => {
+	const prepare = compose( [ normaliseToBlockLevelNodes, gDocs, stripAttributes ] );
```

Comment:
I realized that the `test()` was going to run every time, so if we wrap it into the `run()` we can just have a single function. So for the MarkDown example it would look something like this:

```javascript
export default function( nodes ) {
    if ( ! isMarkDown( nodes ) ) {
        return nodes;
    }
    return convertMarkDownNodes( nodes );
}
```
and that would only run `convertMarkDownNodes()` if the first conditional doesn't do an early return. The `isMarkDown()` function would be whatever the `test()` method was in the original implementation.

So yes they all run every time, but I would expect them to only run if they actually need to.